### DISCUSSION TITLE
parallelism of workflows

### DISCUSSION THREAD

**Discussion by ekawas-td** [argoproj/argo-workflows#8987] [Q&A] [answered]
Hi,

https://github.com/argoproj/argo-workflows/blob/v3.2.6/docs/workflow-controller-configmap.yaml#L24 has the parallelism setting for limiting the number of concurrent workflows by namespace. I was wondering whether there was a way to limit the number by selector or workflow template?

I have seen documentation [here](https://argoproj.github.io/argo-workflows/synchronization/#introduction), specifically:

```apiVersion: v1
kind: ConfigMap
metadata:
 name: my-config
data:
  template: "2"  # Two instances of template can run at a given time in particular namespace
```

but we did not have much luck changing the "2" to anything else and having it limit the number of instances... Anyone know of a way to apply limits to instances of a template?

**Comment by terrytangyuan** [comment, marked as answer]
Template level `parallelism` might be what you are looking for. https://github.com/argoproj/argo-workflows/blob/master/examples/parallelism-template-limit.yaml

**Reply by ekawas-td** [reply]
That definitely sounds like what I am looking for. Does that work with `WorkflowTemplates` too?

**Reply by terrytangyuan** [reply]
You can define it inside a workflowTemplate. If that's what you are asking.
